### PR TITLE
Backport logging-related `AntClassLoader` fixes to our fork

### PR DESCRIPTION
**Pull request by basil**
See [Ant Bug 62764](https://bz.apache.org/bugzilla/show_bug.cgi?id=62764). A straightforward backport of the following commits from Ant 1.10.6 to our forked copy of `AntClassLoader`:

- [commit 189ad7e59](https://github.com/apache/ant/commit/189ad7e59) bz-62764 Add a debug log message when we skip invalid path elements from the `AntClassLoader`'s classpath
- [commit 008f1c8be](https://github.com/apache/ant/commit/008f1c8be) Embrace `StringUtils#getStackTrace`
- [commit 13c01b296](https://github.com/apache/ant/commit/13c01b296) Improve skipping of non-zips

### Proposed changelog entries

Developer: `AntClassloader` will now ignore files that are part of the classpath but not zip files when scanning for resources. It used to throw an exception.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
